pkgs/tools/security/ghidra/lib/ghidra.nix

```diff
@@ -0,0 +1,67 @@
+#TODO It may be possible to use symlinkjoin to add plugins if you write the unwrapped launcher script to not use readlink
+
+# plugins is a list of derivations to symlink into <ghidraroot>/Ghidra/Extensions,
+# this is entirely sufficient for ghidra to operate.
+{
+  stdenv, fetchurl, unzip, makeWrapper, autoPatchelfHook, # Utilities
+  jdk, pam, # Deps
+  lib, config, plugins ? [], extraLaunchers ? {}, # Ghidra
+  pkgs #.lib
+  }:
+
+#Recursive let means these two need to be separate
+let
+  inherit (lib) jdkWrapper installPlugin unpackPlugin writeCustomLauncher;
+in
+let
+  lib = pkgs.lib;
+in
+
+  stdenv.mkDerivation {
+    name = "ghidra-9.0";
+
+    src = fetchurl {
+      url = https://ghidra-sre.org/ghidra_9.0_PUBLIC_20190228.zip;
```

Comment:
Ghidra 9.0.0 has several security vulnerabilities. You may want to update to 9.0.2 while you're at it. IIRC, I did it before and it was not much of a problem, but I would rather not make a pull request now because I don't want to subject you to merge conflicts